pkg/logs/launchers/container/tailerfactory/socket.go

```diff
@@ -43,7 +43,7 @@ func (tf *factory) makeSocketTailer(source *sources.LogSource) (Tailer, error) {
 	// available at some point, so chances are good that tailing will succeed.
 
 	pipeline := tf.pipelineProvider.NextPipelineChan()
-	readTimeout := time.Duration(pkgconfigsetup.Datadog().GetInt("logs_config.docker_client_read_timeout")) * time.Second
+	readTimeout := pkgconfigsetup.Datadog().GetDuration("logs_config.docker_client_read_timeout") * time.Second
```

Comment:
Not if you use `GetDuration` you don't have to multply by seconds anymore as durations can be specified as a string: `30s` `10m` `1h`
[here is an example in our config](https://github.com/DataDog/datadog-agent/blob/71c28e9375faa05560d1a4cfe1b837139e197f16/pkg/config/setup/config.go#L489) and [here is an example](https://github.com/DataDog/datadog-agent/blob/1a4e43dd588520ec34e9a5b648ce84e1153a9b23/comp/networkpath/npcollector/npcollectorimpl/config.go#L45) of how it's used. So to propertly supprot this, you can:
- remove the `* time.Second` and 
- update the default in `config.go` 
- update the config template to call out the correct type and default.